File: src/com/secretary/DatabaseConstants.java
package com.secretary.helloworld;

/**
 * Database constants extracted from TaskDatabaseHelper
 * for better organization and reusability.
 */
public class DatabaseConstants {
    // Database configuration
    public static final String DATABASE_NAME = "taskmaster.db";
    public static final int DATABASE_VERSION = 5;

    // Table names
    public static final String TABLE_TASKS = "tasks";
    public static final String TABLE_COMPLETIONS = "completions";

    // Tasks table columns
    public static final String COLUMN_ID = "id";
    public static final String COLUMN_TITLE = "title";
    public static final String COLUMN_DESCRIPTION = "description";
    public static final String COLUMN_CATEGORY = "category";
    public static final String COLUMN_CREATED_AT = "created_at";
    public static final String COLUMN_DUE_DATE = "due_date";
    public static final String COLUMN_IS_COMPLETED = "is_completed";
    public static final String COLUMN_PRIORITY = "priority";

    // Recurrence columns
    public static final String COLUMN_RECURRENCE_TYPE = "recurrence_type";
    public static final String COLUMN_RECURRENCE_AMOUNT = "recurrence_amount";
    public static final String COLUMN_RECURRENCE_UNIT = "recurrence_unit";
    public static final String COLUMN_LAST_COMPLETED_DATE = "last_completed_date";
    public static final String COLUMN_COMPLETIONS_THIS_PERIOD = "completions_this_period";
    public static final String COLUMN_CURRENT_PERIOD_START = "current_period_start";

    // Streak columns
    public static final String COLUMN_CURRENT_STREAK = "current_streak";
    public static final String COLUMN_LONGEST_STREAK = "longest_streak";
    public static final String COLUMN_LAST_STREAK_DATE = "last_streak_date";

    // Completion history table columns
    public static final String COLUMN_COMPLETION_ID = "completion_id";
    public static final String COLUMN_TASK_ID = "task_id";
    public static final String COLUMN_COMPLETED_AT = "completed_at";
    public static final String COLUMN_TIME_SPENT = "time_spent_minutes";
    public static final String COLUMN_DIFFICULTY = "difficulty";
    public static final String COLUMN_NOTES = "notes";

    // Prevent instantiation
    private DatabaseConstants() {}
}

File: app/src/main/java/com/secretary/helloworld/shared/database/DatabaseConstants.kt
package com.secretary.helloworld.shared.database

/**
 * Database constants extracted from TaskDatabaseHelper
 * for better organization and reusability.
 *
 * Converted to Kotlin in Phase 4.5.3 (Kotlin Migration)
 */
object DatabaseConstants {
    // Database configuration
    const val DATABASE_NAME = "taskmaster.db"
    const val DATABASE_VERSION = 5

    // Table names
    const val TABLE_TASKS = "tasks"
    const val TABLE_COMPLETIONS = "completions"

    // Tasks table columns
    const val COLUMN_ID = "id"
    const val COLUMN_TITLE = "title"
    const val COLUMN_DESCRIPTION = "description"
    const val COLUMN_CATEGORY = "category"
    const val COLUMN_CREATED_AT = "created_at"
    const val COLUMN_DUE_DATE = "due_date"
    const val COLUMN_IS_COMPLETED = "is_completed"
    const val COLUMN_PRIORITY = "priority"

    // Recurrence columns
    const val COLUMN_RECURRENCE_TYPE = "recurrence_type"
    const val COLUMN_RECURRENCE_AMOUNT = "recurrence_amount"
    const val COLUMN_RECURRENCE_UNIT = "recurrence_unit"
    const val COLUMN_LAST_COMPLETED_DATE = "last_completed_date"
    const val COLUMN_COMPLETIONS_THIS_PERIOD = "completions_this_period"
    const val COLUMN_CURRENT_PERIOD_START = "current_period_start"

    // Streak columns
    const val COLUMN_CURRENT_STREAK = "current_streak"
    const val COLUMN_LONGEST_STREAK = "longest_streak"
    const val COLUMN_LAST_STREAK_DATE = "last_streak_date"

    // Completion history table columns
    const val COLUMN_COMPLETION_ID = "completion_id"
    const val COLUMN_TASK_ID = "task_id"
    const val COLUMN_COMPLETED_AT = "completed_at"
    const val COLUMN_TIME_SPENT = "time_spent_minutes"
    const val COLUMN_DIFFICULTY = "difficulty"
    const val COLUMN_NOTES = "notes"
}
